### PR TITLE
Trigger notification on event

### PR DESCRIPTION
**Pull request by armoj**
* Move `sendConfirmationNotification` invocation to the system endpoint handler.
* Update tests

### CODE REVIEW

File: questionnaire/questionnaire-service.js [hunk 1]
```diff
@@ -215,7 +215,6 @@ function createQuestionnaireService({
             // if the caseReferenceNumber exists, then the submission is COMPLETE.
             if (caseReferenceNumber) {
                 await updateQuestionnaireSubmissionStatus(questionnaireId, 'COMPLETED');
```
Comment (sinclairs) [inline questionnaire/questionnaire-service.js:217]: Should this not treated the same as sending a notification? e.g. they'll get a notification eventually, however their application will still show "IN_PROGRESS"

File: questionnaire/routes.js [hunk 2]
```diff
@@ -83,6 +83,7 @@ router
                 'system',
                 answers
             );
+            await questionnaireService.sendConfirmationNotification(req.params.questionnaireId);
```
Comment (sinclairs) [inline questionnaire/routes.js:86]: for a tiny performance bump... does this need to be awaited?

**Review by sinclairs — CHANGES_REQUESTED** [review]

File: questionnaire/routes.js (hunk 2 above)
Comment (armoj) [reply questionnaire/routes.js:86]: nope it doesn't! wasn't originally. don't know why I added it in. Good catch

File: questionnaire/questionnaire-service.js (hunk 1 above)
Comment (armoj) [reply questionnaire/questionnaire-service.js:217]: If you are basically asking: "should we move the updating of the submission_status in to the `system answers` route handler?" then I did think about that yeah. Maybe it should be that way!

**Review by BarryPiccinni — APPROVED** [review]

**Pull request merged**